src/session-pool.ts

```diff
@@ -781,10 +792,13 @@ export class SessionPool extends EventEmitter implements SessionPoolInterface {
     }
 
     try {
+      this._numWaiters++;
```

Comment:
Should we move this line before the check in Line 783? 

E.g., assume that `sessionPool._pending = 1` and `sessionPool._getSession()` is called, when we reach the check in Line 783, we have a pending session and a waiter (the sessionPool._getSession() thread itself should be a waiter and `_numWaiters` is supposed to be `1`). Then, when `_pending + _pendingPrepare` is equal to `_numWaiters`, we should not create a new session. 

What do you think about this logic?

Reply:
I would prefer to have it at line 795, as that is the moment it actually becomes a waiter (i.e. the moment it actually starts waiting). We don't have to worry about other threads calling `sessionPool._getSession()` between lines 783 and 795 because of the single-threaded nature of the event loop in Node.

Reply:
At my first look, I would think `_numWaiters` would include the thread itself, but actually not. It's not a big problem. 

It might be good to say something in the comment that for that `if` check, the current thread is not included in the `_numWaiters` so when `this._pending + this._pendingPrepare === this._numWaiters`, we need to create a new session for the current thread.

Reply:
I added a comment about that to clarify it.